tools/manifest-templator/manifest-templator.go

```diff
@@ -72,40 +77,299 @@ var (
+	// service accounts are represented as a map to prevent us from generating the
+	// same service account multiple times.
+	deployments := []appsv1.Deployment{
+		components.GetDeploymentOperator(operatorParams),
+		components.GetDeploymentWebhook(
+			*operatorNamespace,
+			*webhookImage,
+			"IfNotPresent",
+			*hcoKvIoVersion,
+			[]corev1.EnvVar{},
+		),
+	}
+	// hco-operator and hco-webhook
+	for i := range deployments {
+		overwriteDeploymentLabels(&deployments[i], hcoutil.AppComponentDeployment)
+	}
+
+	services := []v1.Service{
+		components.GetServiceWebhook(*operatorNamespace),
+	}
+
+	serviceAccounts := map[string]v1.ServiceAccount{
+		"hyperconverged-cluster-operator": components.GetServiceAccount(*operatorNamespace),
+	}
+	permissions := make([]rbacv1.Role, 0)
+	roleBindings := make([]rbacv1.RoleBinding, 0)
+	clusterPermissions := []rbacv1.ClusterRole{
+		components.GetClusterRole(),
+	}
+	clusterRoleBindings := []rbacv1.ClusterRoleBinding{
+		components.GetClusterRoleBinding(*operatorNamespace),
+	}
+
+	for _, csvStr := range componentsWithCSVs {
+		if csvStr.Csv == "" {
+			continue
+		}
+		csvBytes := []byte(csvStr.Csv)
+
+		csvStruct := &csvv1alpha1.ClusterServiceVersion{}
+
+		check(yaml.Unmarshal(csvBytes, csvStruct))
+
+		services = getServices(csvStruct, services)
+
+		strategySpec := csvStruct.Spec.InstallStrategy.StrategySpec
+
+		// CSVs only contain the deployment spec, we must wrap
+		// the spec with the Type and Object Meta to make it a valid
+		// manifest
+		for _, deploymentSpec := range strategySpec.DeploymentSpecs {
+			deploy := getBasicDeployment(deploymentSpec)
+			injectWebhookMounts(csvStruct.Spec.WebhookDefinitions, &deploy)
+			overwriteDeploymentLabels(&deploy, csvStr.Component)
+			deployments = append(deployments, deploy)
+		}
+
+		// Every permission we encounter in a CSV means we need to (potentially)
+		// add a service account to our map, add a Role to our slice of permissions
+		// (much like we did with deployments we need to wrap the rules with the
+		// Type and Object Meta to make a valid Role), and add a RoleBinding to our
+		// slice.
+		for _, permission := range strategySpec.Permissions {
+			serviceAccounts[permission.ServiceAccountName] = getBasicServiceAccount(permission)
+			permissions = append(permissions, getRole(permission))
+			roleBindings = append(roleBindings, getRoleBinding(permission))
+		}
+
+		// Same as permissions except ClusterRole instead of Role and
+		// ClusterRoleBinding instead of RoleBinding.
+		for _, clusterPermission := range strategySpec.ClusterPermissions {
+			serviceAccounts[clusterPermission.ServiceAccountName] = createServiceAccount(clusterPermission)
+			clusterPermissions = append(clusterPermissions, createClusterRole(clusterPermission))
+			clusterRoleBindings = append(clusterRoleBindings, createClusterRoleBinding(clusterPermission))
+		}
+	}
+
+	// Write out CRDs and CR
+	writeOperatorCr()
+	// Write out deployments
```

Comment:
This comment is incorrectly placed.

Reply:
Done, except for the spliting the deployment and the services. This method is wtiting to the same file.